src/Fantomas.Tests/LetBindingTests.fs

```diff
@@ -546,4 +546,10 @@ let ``multiple empty lines between equals and expression`` () =
 
 
     ()
-"""
\ No newline at end of file
+"""
+
+[<Test>]
+let ``should preserve return attributes`` () =
+    formatSourceString false """let f x : [<return: MyCustomAttributeThatWorksOnReturns>] int = x""" config
+    |> should equal """let f x: [<return: MyCustomAttributeThatWorksOnReturns>] int = x
```

Comment:
this is failing at the moment with:
```
  Expected string length 65 but was 68. Strings differ at index 11.
  Expected: "...: [<return: MyCustomAttributeThatWorksOnReturns>] int = x\n"
  But was:  "...: [<``return``:MyCustomAttributeThatWorksOnReturns>] int = x\n"
  ------------------^
```

Reply:
I would replace the active pattern in SourceParser.fs to
```fsharp
let (|Attribute|) (a : SynAttribute) =
    let (LongIdentWithDots s) = a.TypeName
    (s, a.ArgExpr, Option.map (fun (i:Ident) -> i.idText) a.Target)
```

Reply:
done